Mathlib/RingTheory/PowerSeries/Comp.lean

```diff
@@ -0,0 +1,274 @@
+/-
+Copyright (c) 2023 Richard M. Hill. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Richard M. Hill
+-/
+import Mathlib.RingTheory.PowerSeries.Basic
+/-!
+# Definitions
+
+Let `R` be a commutative semiring. Give two formal power series `f(X)` and `g(X)` with coefficients
+in `R`, their formal composition, when it exists, is the power series
+
+  `f ( g ( X ))= ∑ₙ fₙ * g^n`
+
+I.e the `d`-th coefficient of the composition is the sum
```

Comment:
Note that the documentation supports LaTeX; in general, you should write either real lean code in ``` `backticks` ```, or math in `` $dollars$ `` / ``$$dollars$$`; pseudo-math/code is arguably the worst of both worlds.

Reply:
thank you.